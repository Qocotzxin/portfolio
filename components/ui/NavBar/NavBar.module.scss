.NavBar {
  width: 100%;
  height: 4rem;
  padding-left: var(--ce-space-8);
  padding-right: var(--ce-space-8);
  background: var(--ce-body-bg);
  display: flex;
  justify-content: space-between;
  position: fixed;
  top: 0;
  left: 0;

  &__block {
    display: flex;
    align-items: center;
    justify-content: center;

    & > * {
      margin-right: var(--ce-space-2);
    }

    &--wide {
      & > * {
        margin-right: var(--ce-space-4);
      }
    }
  }
}
