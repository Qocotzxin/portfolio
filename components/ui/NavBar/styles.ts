import styles from "./NavBar.module.scss";

export const Classes = {
  base: styles.NavBar,
  baseDark: styles["NavBar--dark"],
  block: styles.NavBar__block,
  blockWide: styles["NavBar__block--wide"],
};
