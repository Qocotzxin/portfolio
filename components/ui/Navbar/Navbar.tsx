import {
  Box,
  Flex,
  HStack,
  Link,
  Show,
  VisuallyHidden,
} from "@chakra-ui/react";
import ColorModeSwitcher from "@ui/ColorModeSwitcher";
import NavHeading from "@ui/NavHeading";
import NavLinks from "@ui/NavLinks";
import NavLogo from "@ui/NavLogo";
import NextLink from "next/link";
import { FC } from "react";

const Navbar: FC = () => (
  <Flex
    as="nav"
    w="100%"
    h="16"
    px="12"
    py="2"
    bg="transparent"
    justifyContent="center"
  >
    <Flex
      justifyContent="space-between"
      alignItems="center"
      maxW="80em"
      w="100%"
    >
      <HStack spacing={2}>
        <NextLink href="/" passHref>
          <Link>
            <Show breakpoint="(max-width: 767px)">
              <Box>
                <NavLogo withAnimation />
              </Box>
            </Show>
            <Show above="md">
              <NavHeading />
            </Show>
          </Link>
        </NextLink>
      </HStack>

      <HStack spacing={4} fontSize="2xl" fontWeight="bold">
        <NavLinks />
        <ColorModeSwitcher />
      </HStack>
    </Flex>
  </Flex>
);

export default Navbar;
