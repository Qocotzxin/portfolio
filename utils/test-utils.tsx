import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "@context/hygraphContext";

const AllTheProviders: FC<PropsWithChildren> = ({ children }) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={{}}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
