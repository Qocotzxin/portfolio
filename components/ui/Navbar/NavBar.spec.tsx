import { render } from "@utils/test-utils";
import NavBar from "./Navbar";

describe("NavBar", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});
