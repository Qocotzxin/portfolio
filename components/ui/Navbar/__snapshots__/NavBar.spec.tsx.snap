// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Should match snapshot. 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  height: var(--chakra-sizes-16);
  -webkit-padding-start: var(--chakra-space-12);
  padding-inline-start: var(--chakra-space-12);
  -webkit-padding-end: var(--chakra-space-12);
  padding-inline-end: var(--chakra-space-12);
  padding-top: var(--chakra-space-2);
  padding-bottom: var(--chakra-space-2);
  background: var(--chakra-colors-transparent);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  max-width: 80em;
  width: 100%;
}

.emotion-2 {
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-fast);
  transition-timing-function: var(--chakra-transition-easing-ease-out);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  color: var(--chakra-colors-gray-800);
}

.emotion-2:hover,
.emotion-2[data-hover] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2:focus-visible,
.emotion-2[data-focus-visible] {
  box-shadow: var(--chakra-shadows-outline);
}

.chakra-ui-dark .emotion-2:not([data-theme]),
[data-theme=dark] .emotion-2:not([data-theme]),
.emotion-2[data-theme=dark] {
  color: var(--chakra-colors-gray-50);
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: var(--chakra-fontSizes-2xl);
  font-weight: var(--chakra-fontWeights-bold);
}

.emotion-3>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: var(--chakra-space-4);
  margin-inline-start: var(--chakra-space-4);
}

.emotion-4 {
  -webkit-order: 100;
  -ms-flex-order: 100;
  order: 100;
  width: 64px;
  height: 32px;
  background: var(--chakra-colors-blue-700);
  border-radius: 18px;
  position: relative;
}

.chakra-ui-dark .emotion-4:not([data-theme]),
[data-theme=dark] .emotion-4:not([data-theme]),
.emotion-4[data-theme=dark] {
  border-color: var(--chakra-colors-gray-50);
}

.emotion-5 {
  border-radius: 50%;
  background: var(--chakra-colors-blue-700);
  width: 18px;
  height: 18px;
  position: absolute;
  left: 0px;
  top: 7px;
  z-index: 3;
}

.emotion-6 {
  border-radius: 50%;
  background: var(--chakra-colors-yellow-400);
  width: 18px;
  height: 18px;
  position: absolute;
  left: 6px;
  top: 7px;
  -webkit-transform: translateX(0px);
  -moz-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  -webkit-transition: -webkit-transform 0.5s;
  transition: transform 0.5s;
  z-index: 2;
}

.emotion-7 {
  width: 16px;
  height: 16px;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: var(--chakra-colors-white);
  position: absolute;
  right: 8px;
  top: 7px;
  opacity: 1;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  z-index: 1;
}

.emotion-8 {
  width: 16px;
  height: 16px;
  display: inline-block;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: var(--chakra-colors-white);
  position: absolute;
  left: 8px;
  top: 7px;
  opacity: 0;
  -webkit-transition: opacity 0.4s;
  transition: opacity 0.4s;
  z-index: 4;
}

<div>
  <nav
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        class="chakra-link emotion-2"
        href="/"
      />
      <div
        class="chakra-stack emotion-3"
      >
        <button
          class="emotion-4"
        >
          <div
            class="emotion-5"
          />
          <div
            class="emotion-6"
          />
          <svg
            class="chakra-icon stars emotion-7"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.657 6.247c.11-.33.576-.33.686 0l.645 1.937a2.89 2.89 0 0 0 1.829 1.828l1.936.645c.33.11.33.576 0 .686l-1.937.645a2.89 2.89 0 0 0-1.828 1.829l-.645 1.936a.361.361 0 0 1-.686 0l-.645-1.937a2.89 2.89 0 0 0-1.828-1.828l-1.937-.645a.361.361 0 0 1 0-.686l1.937-.645a2.89 2.89 0 0 0 1.828-1.828l.645-1.937zM3.794 1.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387A1.734 1.734 0 0 0 4.593 5.69l-.387 1.162a.217.217 0 0 1-.412 0L3.407 5.69A1.734 1.734 0 0 0 2.31 4.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387A1.734 1.734 0 0 0 3.407 2.31l.387-1.162zM10.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732L9.1 2.137a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L10.863.1z"
            />
          </svg>
          <svg
            class="chakra-icon cloud emotion-8"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.473 9a4.5 4.5 0 0 0-8.72-.99A3 3 0 0 0 3 14h8.5a2.5 2.5 0 1 0-.027-5z"
            />
            <path
              d="M14.544 9.772a3.506 3.506 0 0 0-2.225-1.676 5.502 5.502 0 0 0-6.337-4.002 4.002 4.002 0 0 1 7.392.91 2.5 2.5 0 0 1 1.17 4.769z"
            />
          </svg>
        </button>
      </div>
    </div>
  </nav>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
